.github/workflows/prepare-for-minor-release.yml

```diff
@@ -0,0 +1,171 @@
+name: Prepare repository for new minor release
+run-name: Update az_quickstart repository for new `${{ inputs.release_branch_name }}` minor release branch.
+on:
+  workflow_dispatch:
+    inputs:
+      release_branch_name:
+        description: The name of the new release branch (e.g. 2.8.x) to create.
+        required: true
+jobs:
+  print_inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Print inputs
+        run: |
+          echo "release_branch_name: ${{ inputs.release_branch_name }}"
+
+  branch_name_prep:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    outputs:
+      next_release_branch_alias_constraint: ${{ steps.alias-version-constraint.outputs.NEXT_RELEASE_VERSION_CONSTRAINT }}
+      main_branch_alias: ${{ steps.alias-main-branch.outputs.MAIN_BRANCH_ALIAS }}
+    steps:
+      - name: Create version contstraint string for next release branch alias string.
+        id: alias-version-constraint
+        run: |
+          NEXT_RELEASE_VERSION_CONSTRAINT=$(echo ${{ inputs.release_branch_name }} | sed 's/\(.*\)\.x/~\1/')
+          echo "NEXT_RELEASE_VERSION_CONSTRAINT=${NEXT_RELEASE_VERSION_CONSTRAINT}" >> $GITHUB_OUTPUT
+          echo "Next release version constraint: $NEXT_RELEASE_VERSION_CONSTRAINT" >> $GITHUB_STEP_SUMMARY
+      - name: Create the composer main branch alias string.
+        id: alias-main-branch
+        run: |
+          MAIN_BRANCH_ALIAS="${{ inputs.release_branch_name }}-dev"
+          echo "MAIN_BRANCH_ALIAS=${MAIN_BRANCH_ALIAS}" >> $GITHUB_OUTPUT
+          echo "Main branch alias: $MAIN_BRANCH_ALIAS" >> $GITHUB_STEP_SUMMARY
+
+  dispatch:
+    needs: branch_name_prep
+    strategy:
+      matrix:
+        repo:
+          - az-digital/az-quickstart-scaffolding
+    runs-on: ubuntu-latest
+    steps:
+      - name: Notify dependencies
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.REPO_DISPATCH_TOKEN }}
+          repository: ${{ matrix.repo }}
+          event-type: az_quickstart_prep_for_minor_release
+          client-payload: '{"release_branch_name": "${{ inputs.release_branch_name }}" , "next_release_branch_alias_constraint": "${{needs.branch_name_prep.outputs.next_release_branch_alias_constraint}}", "main_branch_alias": "${{needs.branch_name_prep.outputs.main_branch_alias}}"}'
+
+  prepare_for_new_minor_release:
+    name: Prepare for a new minor release
+    runs-on: ubuntu-latest
+    needs: branch_name_prep
+    steps:
+    - name: Checkout az_quickstart
+      uses: actions/checkout@v4
+      with:
+        repository: az-digital/az_quickstart
+        ref: main
+
+    - name: Create and check out release branch
+      run: |
+        git checkout -b ${{ inputs.release_branch_name }}
+
```

Comment:
Should we push the release branch up before creating a PR against it, or should we create a release branch and then change it via PR after the fact?